sqs_workers/memory_sqs.py

```diff
@@ -155,22 +158,47 @@ def delete_messages(self, Entries):
         See: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/
              services/sqs.html#SQS.Queue.delete_messages
         """
-        message_ids = {entry["Id"] for entry in Entries}
+        found_entries = []
+        not_found_entries = []
 
-        successfully_deleted = set()
-        push_back_messages = []
+        for e in Entries:
+            if e["Id"] in self.in_flight:
+                found_entries.append(e)
+                self.in_flight.pop(e["Id"])
+            else:
+                not_found_entries.append(e)
 
-        for message in self.messages:
-            if message.message_id in message_ids:
-                successfully_deleted.add(message.message_id)
+        return {
+            "Successful": [{"Id": e["Id"]} for e in found_entries],
+            "Failed": [{"Id": e["Id"]} for e in not_found_entries],
+        }
+
+    def change_message_visibility_batch(self, Entries):
```

Comment:
I wonder whether we could make use of this in Todoist, where we do `message.change_visibility()` in a couple of places:

https://github.com/Doist/Todoist/blob/9268b1ea3114307c3c4e9e81d361e1db4994d6ca/todoist/sqs_workers/custom_processors.py#L57-L59

And:

https://github.com/Doist/Todoist/blob/9268b1ea3114307c3c4e9e81d361e1db4994d6ca/todoist/sqs_workers/custom_processors.py#L83-L85

Reply:
Oh, that's a good point! I will check it out!